src/builder/BridgeRoutes.sol

```diff
@@ -0,0 +1,465 @@
+                    paymentAssetNeeded += transferAmount;
+                }
+                if (paymentAssetBalanceOnChain < paymentAssetNeeded) {
+                    revert MaxCostTooHigh();
+                }
+            }
+        }
+
+        // FUNDS_UNAVAILABLE
+        // For some reason, funds that may otherwise be bridgeable or held by the user cannot be made available to fulfill the transaction.
+        // Funds cannot be bridged, e.g. no bridge exists
+        // Funds cannot be withdrawn from comet, e.g. no reserves
+        // In order to consider the availability here, we’d need comet data to be passed in as an input. (So, if we were including withdraw.)
+        uint256 aggregateTransferAssetAvailableBalance;
+        for (uint i = 0; i < transferAssetPositions.length; ++i) {
+            uint256 srcChainId = transferAssetPositions[i].chainId;
+            for (uint j = 0; j < transferAssetPositions[i].accountBalances.length; ++j) {
+                if (hasBridge(srcChainId, chainId, assetSymbol)) {
+                    aggregateTransferAssetAvailableBalance += transferAssetPositions[i].accountBalances[j].balance;
+                }
+            }
+        }
+        if (aggregateTransferAssetBalance < amount) {
+            revert FundsUnavailable();
+        }
+
+        // Construct Quark Operations:
+
+        // If Payment.isToken:
+            // Wrap Quark operation around a Paycall/Quotecall
+            // Process for generating Paycall transaction:
+
+        // We need to find the (payment token address, payment token price feed address) to derive the CREATE2 address of the Paycall script
+        // TODO: define helper function to get (payment token address, payment token price feed address) given a chain ID
+
+        // TODO:
+        // If not enough assets on the chain ID:
+            // Then bridging is required AND/OR withdraw from Comet is required
+            // Prepend a bridge action to the list of actions
+            // Bridge `amount` of `chainAsset` to `recipient`
+        QuarkOperation memory bridgeQuarkOperation;
+        // TODO: implement get assetBalanceOnChain
+        uint256 transferAssetBalanceOnTargetChain = getAssetBalanceOnChain(assetSymbol, chainId, chainAccountsList);
+        // Note: User will always have enough payment token on destination chain, since we already check that in the MaxCostTooHigh() check
+        if (transferAssetBalanceOnTargetChain < amount) {
+            // Construct bridge operation if not enough funds on target chain
+            // TODO: bridge routing logic (which bridge to prioritize, how many bridges?)
+
+            // TODO: construct action contexts
+            if (payment.isToken) {
+                // wrap around paycall
+            } else {
+                address scriptAddress = getCodeAddress(codeJar, type(BridgeActions).creationCode);
+                bridgeQuarkOperation = QuarkOperation({
+                    nonce: , // TODO: get next nonce
```

Comment:
this can be derived from `chainAccountsList[*].quarkStates`.

Reply:
Yeah, though this is block of code is not supposed to be in `BridgeRoutes` (see https://github.com/compound-finance/quark-scripts/pull/3#discussion_r1629880033). The latest commit no longer has this diff.